File: src/MuscleCS/TwinModel/TwinStructure.cs

namespace MuscleCore.TwinModel
{
    public class TwinStructure
    {
        public TwinNodes Nodes { get; set; }
        public TwinElements Elements { get; set; }
        public TwinActions Additional { get; set; } // Additional actions (Loads and Prestress) on the structure
        public TwinActions Applied { get; set; } // Already applied actions (Loads and Prestress) on the structure
        public TwinNodesResults InitialNodesResults { get; set; } // Previous nodes results (from the already applied actions)
        public TwinElementsResults InitialElementsResults { get; set; } // Previous elements results (from the already applied actions)
        public TwinStructure()
        {
            Nodes = new TwinNodes();
            Elements = new TwinElements();
            Additional = new TwinActions();
            Applied = new TwinActions();
            InitialNodesResults = new TwinNodesResults();
            InitialElementsResults = new TwinElementsResults();
        }
        public TwinStructure(TwinNodes nodes, TwinElements elements)
        {
            Nodes = nodes;
            Elements = elements;
            Additional = new TwinActions();
            Applied = new TwinActions();
            InitialNodesResults = new TwinNodesResults();
            InitialElementsResults = new TwinElementsResults();
        }

        public TwinStructure(TwinNodes nodes, TwinElements elements, TwinActions additional)
        {
            Nodes = nodes;
            Elements = elements;
            Additional = additional;
            Applied = new TwinActions();
            InitialNodesResults = new TwinNodesResults();
            InitialElementsResults = new TwinElementsResults();
        }

        public TwinStructure(TwinNodes nodes, TwinElements elements, TwinActions additional, TwinActions applied, TwinNodesResults initialNodesResults, TwinElementsResults initialElementsResults)
        {
            Nodes = nodes;
            Elements = elements;
            Additional = additional;
            Applied = applied;
            InitialNodesResults = initialNodesResults;
            InitialElementsResults = initialElementsResults;
        }
    }
}

File: src/MusclePy/twin_model/twin_structure.py

import numpy as np


class Twin_Structure:
    def __init__(self, nodes=None, elements=None, additional=None, applied=None, initial_nodes_results=None, initial_elements_results=None):
        """Python equivalent of C# Twin_Structure class"""
        self.nodes = Twin_Nodes(nodes)  if nodes is not None else Twin_Nodes()
        self.elements = Twin_Elements(elements) if elements is not None else Twin_Elements()

        # Additional actions (Loads and Prestress) on the structure
        self.additional = Twin_Actions(additional) if additional is not None else Twin_Actions()

        # Already applied actions (Loads and Prestress) on the structure  
        self.applied = Twin_Actions(applied) if applied is not None else Twin_Actions() 

        # Previous nodes results (from the already applied actions) 
        self.initial_nodes_results = Twin_NodesResults(initial_nodes_results) if initial_nodes_results is not None else Twin_NodesResults()

        # Previous elements results (from the already applied actions) 
        self.initial_elements_results = Twin_ElementsResults(initial_elements_results) if initial_elements_results is not None else Twin_ElementsResults()

